app/models/account.rb

```diff
@@ -85,7 +81,12 @@ def self.update_if_needed!(account, user)
   # @return [Hash]
   #
   def self.account_attrs_from_user(user)
-    { idme_uuid: user.uuid, edipi: user.edipi, icn: user.icn, sec_id: user.sec_id }
+    {
+      idme_uuid: user.idme_uuid,
```

Comment:
mostly just for my benefit while reviewing - None of these are attributes stored on user.  

      idme_uuid: user.idme_uuid, - > delegated to identity
      sec_id: user.sec_id, -> identity.sec_id || Mvi.for_user(self).profile&.sec_id  (mvi is cached user.user.uuid)
      edipi: user.edipi, - identity.edipi ||  mvi&.edipi
      icn: user.icn -> delegated to mvi

Reply:
and then, we should have all of those things available because the objects are found/created/instantiaed in this order:

-   Find User. if exists using uuid from saml (existing_user)
-   create UserIdentity from SAML attributes
-   Create User. uuid: @user_identity.attributes[:uuid]
-   Create Session. uuid: @user.uuid
-   In a job: find or create Account using User.uuid

Reply:
@annaswims based on that logic, it probably make sense to parse the ICN number out of the `va_eauth_gcIds` attribute on an SSOe response, and update the `user.icn` function to use that if possible.  That way we wouldn't need to query MVI to create an Account when a user logs in via SSOe.

Reply:
ooh, that'd be nice!

Reply:
Agreed, I think this could be an optimization for a future ticket, and even starts laying the groundwork for moving away from the separate MVI lookups altogether.

Reply:
created an implementation issue for this https://github.com/department-of-veterans-affairs/va.gov-team/issues/7424